pkg/controllers/state/cluster.go

```diff
@@ -461,18 +465,27 @@ func (c *Cluster) cleanupNode(name string) {
 }
 
 func (c *Cluster) populateStartupTaints(ctx context.Context, n *StateNode) error {
+	if n.startupTaintsInitialized {
+		return nil
+	}
+
 	if nodeclaimutil.OwnerKey(n).Name == "" {
 		return nil
 	}
 	owner, err := nodeclaimutil.Owner(ctx, c.kubeClient, n)
 	if err != nil {
 		return client.IgnoreNotFound(err)
 	}
+	n.startupTaintsInitialized = true
 	n.startupTaints = owner.Spec.Template.Spec.StartupTaints
 	return nil
 }
 
 func (c *Cluster) populateInflight(ctx context.Context, n *StateNode) error {
+	if n.inflightInitialized {
```

Comment:
Can you add a comment here why we do this? 
```suggestion
        // We only need to populate the in-flight details once. This prevents us from firing many un-needed errors when the cloud provider Instance types change.
	if n.inflightInitialized {
```